app/Main.hs

```diff
@@ -120,17 +124,24 @@ main = do
               CPRPathNotFound path -> pathNotFound path
               CPRMissingEntryName -> printError
                 "The destination path is not a valid entry path. Please specify the new name of the entry."
-              CPRDestinationIsDirectory paths -> do
-                let header = "The following entries cannot be renamed because a directory already exists at the destination."
-                let errorMsgs = NE.toList paths <&> \(from, to) -> "Cannot rename '" +| from |+ "' to '" +| to |+ "'."
-                printError $ unlinesF @_ @Builder $ header : "" : errorMsgs
-              CPREntryAlreadyExists paths -> do
-                let header = unlinesF @_ @Builder
-                      [ "The following entries cannot be renamed because an entry already exists at the destination."
-                      , "Use '--force' or '-f' to overwrite existing entries."
-                      ]
-                let errorMsgs = NE.toList paths <&> \(from, to) -> "Cannot rename '" +| from |+ "' to '" +| to |+ "'."
-                printError $ unlinesF @_ @Builder $ header : "" : errorMsgs
+              CPRCreateErrors errors -> do
+                errorMsgs <- forM errors \case
+                              (from, CEEntryAlreadyExists entryTo) -> do
+                                let header = "'" +| from |+ "' to '" +| entryTo ^. path |+ "':"
+                                let errorMsg = unlinesF @_ @Builder
+                                      [ "An entry already exists at '" +| entryTo ^. path |+ "'."
+                                      , "Use '--force' or '-f' to overwrite existing entries."
+                                      ]
+                                pure $ unlinesF @_ @Builder $ header : [indentF 2 errorMsg]
+                              (from, CEDestinationIsDirectory entryTo) -> do
+                                let header = "'" +| from |+ "' to '" +| entryTo ^. path |+ "':"
+                                let errorMsg = "'" +| entryTo ^. path |+ "' is a directory."
+                                pure $ unlinesF @_ @Builder $ header : [indentF 2 errorMsg]
+                              (from, CEParentDirectoryIsEntry (entryTo, clashed)) -> do
+                                let header = "'" +| from |+ "' to '" +| entryTo ^. path |+ "':"
+                                let errorMsg = "Attempted to create the directory '" +| clashed |+ "' but an entry exists at that path."
+                                pure $ unlinesF @_ @Builder $ header : [indentF 2 errorMsg]
```

Comment:
All 3 branches follow the exact same template for building the message, the only thing that changes is the `errorMsg`. So maybe this could be refactored to avoid this duplication.

Reply:
Doesn't look like it's done :/ Also see my comment above about having a function dedicated to building the `errorMsg` part, and reusing that function for `create` + `copy/rename`.